src/zenml/integrations/kubernetes/orchestrators/kubernetes_orchestrator_entrypoint.py

```diff
@@ -652,6 +664,12 @@ def check_job_status(node: Node) -> NodeStatus:
                     error_message,
                 )
                 return NodeStatus.FAILED
+            elif is_node_heartbeat_unhealthy(node):
+                logger.error(
+                    "Heartbeat for step `%s` indicates unhealthy status.",
+                    step_name,
+                )
+                return NodeStatus.FAILED
```

Comment:
If we get to this point, it means the Kubernetes job is still actively running but for some reason the heartbeat stopped, correct? In that case, it would probably be a good idea to also stop the job from continuing?